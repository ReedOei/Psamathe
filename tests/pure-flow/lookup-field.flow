;

{ l : ! bool |-> true, r : ! bool |-> false } --> var c : { l : ! bool, r : ! bool }

c.l --> var d : bool

