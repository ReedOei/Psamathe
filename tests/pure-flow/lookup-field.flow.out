<generatedTop>
  <common>
    <typeEnv>
      .Map
    </typeEnv>
    <types>
      .Map
    </types>
    <freshCounter>
      7
    </freshCounter>
  </common>
  <k>
    .
  </k>
  <functions>
    .Map
  </functions>
  <lookup>
    c |-> loc ( 0 )
    d |-> loc ( 6 )
  </lookup>
  <storage>
    loc ( 0 ) |-> resource ( table ( .Vars ) { l : ! bool , r : ! bool , .VarDefs } , ListItem ( l |-> loc ( 3 )
    r |-> loc ( 5 ) ) )
    loc ( 1 ) |-> resource ( table ( .Vars ) { l : ! bool , r : ! bool , .VarDefs } , .List )
    loc ( 2 ) |-> resource ( bool , .List )
    loc ( 3 ) |-> resource ( bool , .List )
    loc ( 4 ) |-> resource ( bool , .List )
    loc ( 5 ) |-> resource ( bool , ListItem ( false ) )
    loc ( 6 ) |-> resource ( bool , ListItem ( true ) )
  </storage>
  <catchStack>
    .List
  </catchStack>
</generatedTop>
